src/main/scala/ignition/core/cache/ExpiringMultiLevelCache.scala

```diff
@@ -17,9 +17,23 @@ import scala.util.control.NonFatal
 import scala.util.{Failure, Success, Try}
 
 object ExpiringMultiLevelCache {
-  case class TimestampedValue[V](date: DateTime, value: V) {
-    def hasExpired(ttl: FiniteDuration, now: DateTime): Boolean = {
-      date.plus(ttl.toMillis).isBefore(now)
+  case class TimestampedValue[V](date: DateTime,
+                                 value: Option[V] = None,
+                                 hasError: Boolean = false,
+                                 throwable: Option[Throwable] = None) {
```

Comment:
I'd change this class constructor to:

```scala
case class TimestampedValue[V](date: DateTime, value: Try[V])
```

- I think you don't need the hasError flag.
- Try[V] can be a actual value or a throwable, so you don't need the extra "throwable" property.

Reply:
@fernandors87 All those things you pointed out can be solved with this `Try[V]`. Thank you.